Building C library fails due to LR & PC registers confliction

- crosstool-NG version: 1.24.0.167_e7da850
- host OS and its version: Ubuntu 20.04 LTS
- `.config`: [config.txt](https://github.com/crosstool-ng/crosstool-ng/files/5322947/config.txt)
- `build.log`: [build.log](https://github.com/crosstool-ng/crosstool-ng/files/5322948/build.log)
- Target: cortex-m7 (only supports thumb instruction-set and It's an MMU-less architecture)

[ERROR]      libc/sysdeps/linux/arm/setcontext.S:63: Error: LR and PC should not both be in register list -- `ldmia r14,{r14,pc}'
[ERROR]      make[2]: *** [Makerules:372: libc/sysdeps/linux/arm/setcontext.os] Error 1
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'

--- a/scripts/build/arch/arm.sh
+++ b/scripts/build/arch/arm.sh
@@ -70,6 +70,7 @@ CT_DoArchUClibcConfig() {
                 ;;
             thumb)
                 CT_KconfigEnableOption "COMPILE_IN_THUMB_MODE" "${cfg}"
+                CT_KconfigDisableOption "UCLIBC_HAS_CONTEXT_FUNCS" "${cfg}"
                 ;;
         esac
         # FIXME: CONFIG_ARM_OABI does not exist in neither uClibc/uClibc-ng

